### ISSUE TITLE
Zappa should fail if cannot download dependency packages

### ISSUE THREAD

**Issue by ceefour** [Miserlou/Zappa#1235]
## Context
Dependencies (such as psycopg2) is needed by the app. If Zappa cannot download or get it from cache, then update/deploy should fail.

## Expected Behavior
Failed deploy.

## Actual Behavior
The psycopg2 download timed out, then continue deploying a broken app.

## Steps to Reproduce

1. `zappa update prd`

(make the network unstable to fail a dependency, but then stable enough to continue deploying the app)

## Your Environment

Zappa x
Python 3.6 in Anaconda Windows 10 64-bit

* Zappa version used: 0.45.1
* Operating System and Python version: python 3.6.3 anaconda windows 10 64-bit
* The output of `pip freeze`:

```
argcomplete==1.9.2
base58==0.2.4
boto3==1.4.7
botocore==1.7.43
certifi==2017.11.5
chardet==3.0.4
click==6.7
Django==1.11.7
docutils==0.14
durationpy==0.5
future==0.16.0
hjson==3.0.1
idna==2.6
jmespath==0.9.3
kappa==0.6.0
lambda-packages==0.19.0
placebo==0.8.1
psycopg2==2.7.3.2
python-dateutil==2.6.1
python-slugify==1.2.4
pytz==2017.3
PyYAML==3.12
requests==2.18.4
s3transfer==0.1.11
six==1.11.0
toml==0.9.3
tqdm==4.19.1
troposphere==2.0.2
Unidecode==0.4.21
urllib3==1.22
Werkzeug==0.12
wsgi-request-logger==0.4.6
zappa==0.45.1
```

* Link to your project (optional):
* Your `zappa_settings.py`: 

```
{
    "prd": {
        "aws_region": "us-east-1",
        "django_settings": "matagarudabot.settings",
        "profile_name": "default",
        "project_name": "matagarudabot",
        "runtime": "python3.6",
        "s3_bucket": "samaraedu-code",
        "memory_size": 128,
        "keep_warm": true
    }
}
```

**Comment by Miserlou** [comment]
Yep, I think this is a good idea. Does it need a configuration/override option?

PR welcome!

**Comment by ceefour** [comment]
I can't imagine any reason why I want a broken deploy instead of failing. Since an update may take some time I may want to leave it for coffee etc. I wouldn't want to return to see that the deploy has broken the app.

But even better would be auto-retry... If it can retry failed downloads, for, say, 3 times, that would be awesome.

**Comment by ceefour** [comment]
Duplicate of #1040.
